tests/test_instcat_parser.py

```diff
@@ -147,99 +150,145 @@ def test_object_extraction_stars(self):
+        index = np.argsort(id_arr)
+        index = index[np.where(np.in1d(id_arr[index], truth_data['uniqueId']))]
+        np.testing.assert_array_equal(truth_data['uniqueId'], id_arr[index])
 
         ######## test that pupil coordinates are correct to within
         ######## half a milliarcsecond
-
-        x_pup_test, y_pup_test = _pupilCoordsFromRaDec(truth_data['raJ2000'],
-                                                       truth_data['decJ2000'],
-                                                       pm_ra=truth_data['pmRA'],
-                                                       pm_dec=truth_data['pmDec'],
-                                                       v_rad=truth_data['v_rad'],
-                                                       parallax=truth_data['parallax'],
-                                                       obs_metadata=obs_md)
-
-        for gs_obj in gs_object_arr:
-            i_obj = np.where(truth_data['uniqueId'] == gs_obj.uniqueId)[0][0]
-            dd = np.sqrt((x_pup_test[i_obj]-gs_obj.xPupilRadians)**2 +
-                         (y_pup_test[i_obj]-gs_obj.yPupilRadians)**2)
-            dd = arcsecFromRadians(dd)
-            self.assertLess(dd, 0.0005)
+        if 0:
+            # XXX: This test required lsst.sims.  Not sure if we still need it?
+
+            x_pup_test, y_pup_test = _pupilCoordsFromRaDec(truth_data['raJ2000'],
+                                                           truth_data['decJ2000'],
+                                                           pm_ra=truth_data['pmRA'],
+                                                           pm_dec=truth_data['pmDec'],
+                                                           v_rad=truth_data['v_rad'],
+                                                           parallax=truth_data['parallax'],
+                                                           obs_metadata=obs_md)
+
+            for gs_obj in gs_object_arr:
+                i_obj = np.where(truth_data['uniqueId'] == gs_obj.uniqueId)[0][0]
+                dd = np.sqrt((x_pup_test[i_obj]-gs_obj.xPupilRadians)**2 +
+                             (y_pup_test[i_obj]-gs_obj.yPupilRadians)**2)
+                dd = arcsecFromRadians(dd)
+                self.assertLess(dd, 0.0005)
+
+        ######## test that positions are consistent
+
+        for det_name in all_wcs:
+            wcs = all_wcs[det_name]
+            cat = all_cats[det_name]
+            for i in range(cat.nobjects):
+                image_pos = cat.image_pos[i]
+                world_pos = cat.world_pos[i]
+                self.assertLess(world_pos.distanceTo(wcs.toWorld(image_pos)), 0.0005*galsim.arcsec)
+
+        ra_arr = np.array([pos.ra.rad for cat in all_cats.values() for pos in cat.world_pos])
+        dec_arr = np.array([pos.dec.rad for cat in all_cats.values() for pos in cat.world_pos])
+        # XXX: These are only within 1.e-4 radians, which is kind of a lot.
```

Comment:
Almost all of the tests in this file were written by Scott, so I think we should just update things in a way that makes sense to us.

Reply:
I'm hoping @jmeyers314 can take a look at this test to see whether the previous version did make sense, and the failures are a sign of a bug or if the test isn't quite relevant anymore given the new wcs handling.

Reply:
I'll take a closer look.  Browsing `main`, it looks like these positions weren't previously tested at all?  Is that right?  (So maybe they were always 1e-4 rad off?)

Reply:
They weren't tested on main (since we weren't running these tests), but I'm pretty sure they were tested in v1.0.  That was with the older wcs code, not the BatoidWCS.

Reply:
I think that is right, it used the sims code with the distortion model from for the WCS as we discussed before.